reach/reach.py

```diff
@@ -628,6 +628,48 @@ def _most_similar_batch(
                 word_index = np.flip(np.argsort(sims_for_word))
                 yield [(self.indices[indices[idx]], sims_for_word[idx]) for idx in word_index]
 
+    def indices_threshold(
+        self,
+        vectors: npt.NDArray,
+        threshold: float = 0.5,
+        batch_size: int = 100,
+        show_progressbar: bool = False,
+    ) -> Iterator[np.ndarray]:
+        """
+        Retrieve the unsorted indices of the nearest neighbors for the given vectors above a threshold.
+
+        This method is useful if you want to find the indices of the most similar items above a certain threshold
+        but don't need them to be sorted, which is a time consuming operation.
```

Comment:
this is interesting. In the end, I'm not sure if it was the sorting. I'm fine with it as is, but I think the most time was spent on linking indices to names.